CriticalMass/ChatInputView.swift

```diff
@@ -11,109 +11,54 @@ protocol ChatInputDelegate: AnyObject {
     func didTapSendButton(text: String)
 }
 
-class ChatInputView: UIView, UITextFieldDelegate {
-    struct Constants {
-        static let textFieldHeight = CGFloat(46.0)
-        static let sendButtonWidth = CGFloat(65.0)
+class ChatInputView: UIView, IBConstructable {
+    private enum Constants {
         static let messageTextFieldInsets = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
     }
 
     @objc
     dynamic var sendMessageButtonColor: UIColor? {
         willSet {
-            sendButton.setTitleColor(newValue, for: .normal)
-            sendButton.setTitleColor(newValue?.withAlphaComponent(0.4), for: .disabled)
-            sendButton.setTitleColor(newValue?.withAlphaComponent(0.4), for: .highlighted)
+            sendButton.tintColor = newValue
+        }
+    }
+    @objc
+    dynamic var sendMessageButtonBGColor: UIColor? {
+        willSet {
+            sendButton.backgroundColor = newValue
+            var highLightBackgroundColor: UIColor {
+                return newValue == .white ? .lightGray : .darkGray
```

Comment:
Comparing UIColor objects might break two UIColors might result in the same color but not be equal UIColor objects. Adding a specific color for the state might be safer and easier to maintain

Reply:
I knew this would come back at me 😀 I would like an opinion from @zutrinken which grey would fit for both themes